File: services/user-service/internal/dto/response.go
package dto

import db "github.com/rijum8906/go-micro-service/services/user-service/internal/db/generated"

// BaseSuccessResponse represents a consistent API response envelope.
type BaseSuccessResponse[T any] struct {
	Success bool   `json:"success"`
	Message string `json:"message,omitempty"`
	Data    T      `json:"data,omitempty"`
}

// BaseErrorResponse represents a consistent API response envelope.
type BaseErrorResponse struct {
	Success bool                `json:"success"`
	Message string              `json:"message,omitempty"`
	Errors  []BaseResponseError `json:"errors,omitempty"`
}

// BaseResponseError represents a single validation or domain error.
type BaseResponseError struct {
	Field   string `json:"field,omitempty"`
	Message string `json:"message"`
}

type Token struct {
	AccessToken  string `json:"accessToken"`
	RefreshToken string `json:"refreshToken"`
}

type AuthResponse struct {
	Account  *db.Account   `json:"account"`
	Profiles *[]db.Profile `json:"profiles"`
	Token    *Token        `json:"token"`
}

File: services/web/src/types/response.ts
// @/types/response.ts

import type { Token } from './auth';
import type { Account, Profile } from './models';
import type { GetProfileResult } from './result';

export interface ErrorDetail {
  field: string;
  message: string;
}

export interface BaseResponse {
  success: boolean;
  message: string;
}

export interface BaseSuccessResponse<T = unknown> {
  success: true;
  message: string;
  data?: T;
}

export interface BaseErrorResponse {
  success: false;
  message: string;
  errors?: ErrorDetail[];
}

// Matches your Go BaseErrorResponse
export interface ErrorResponse extends BaseResponse {
  errors?: ErrorDetail[];
}

// Matches your Go BaseSuccessResponse[*dto.AuthResponse]
export interface AuthResponse extends BaseResponse {
  data: {
    account: Account;
    profiles: Profile[];
    token: Token;
  };
}

export interface UpdateProfileResponse extends BaseResponse {
  data: Profile;
}

export interface GetProfileResponse extends BaseResponse {
  data: GetProfileResult;
}
